From bbe0c0678e2b998354bbf46c60f376eb3a7aa96b Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 28 Feb 2012 12:53:17 +0100
Subject: [PATCH 4/7] Drop obsolete nographic timer

RH-Author: Paolo Bonzini <pbonzini@redhat.com>
Message-id: <1330433598-21534-5-git-send-email-pbonzini@redhat.com>
Patchwork-id: 37671
O-Subject: [RHEL 6.3 qemu-kvm PATCH 4/5] Drop obsolete nographic timer
Bugzilla: 796575
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Markus Armbruster <armbru@redhat.com>

From: Jan Kiszka <jan.kiszka@siemens.com>

We flush coalesced MMIO in the device models now, and VNC - for which
this was once introduced - is also fine without it as it has its own
refresh timer.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
Signed-off-by: Blue Swirl <blauwirbel@gmail.com>
(cherry picked from commit 9870a5e6cd3df584328728b3d822b23cee28c561)

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---
 vl.c |   12 ------------
 1 files changed, 0 insertions(+), 12 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 vl.c |   12 ------------
 1 files changed, 0 insertions(+), 12 deletions(-)

diff --git a/vl.c b/vl.c
index 9ff9910..ff5ea9b 100644
--- a/vl.c
+++ b/vl.c
@@ -294,7 +294,6 @@ static int icount_time_shift;
 static int64_t qemu_icount_bias;
 static QEMUTimer *icount_rt_timer;
 static QEMUTimer *icount_vm_timer;
-static QEMUTimer *nographic_timer;
 
 uint8_t qemu_uuid[16];
 
@@ -3216,13 +3215,6 @@ static void gui_update(void *opaque)
     qemu_mod_timer(ds->gui_timer, interval + qemu_get_clock(rt_clock));
 }
 
-static void nographic_update(void *opaque)
-{
-    uint64_t interval = GUI_REFRESH_INTERVAL;
-
-    qemu_mod_timer(nographic_timer, interval + qemu_get_clock(rt_clock));
-}
-
 struct vm_change_state_entry {
     VMChangeStateHandler *cb;
     void *opaque;
@@ -6382,10 +6374,6 @@ int main(int argc, char **argv, char **envp)
         }
         dcl = dcl->next;
     }
-    if (ds->gui_timer == NULL) {
-        nographic_timer = qemu_new_timer(rt_clock, nographic_update, NULL);
-        qemu_mod_timer(nographic_timer, qemu_get_clock(rt_clock));
-    }
     text_consoles_set_display(ds);
 
     if (qemu_opts_foreach(&qemu_mon_opts, mon_init_func, NULL, 1) != 0)
-- 
1.7.7.6

